gspread/utils.py

```diff
@@ -43,69 +45,60 @@
 URL_KEY_V1_RE = re.compile(r"key=([^&#]+)")
 URL_KEY_V2_RE = re.compile(r"/spreadsheets/d/([a-zA-Z0-9-_]+)")
 
-Dimension = namedtuple("Dimension", ["rows", "cols"])("ROWS", "COLUMNS")
-ValueRenderOption = namedtuple(
-    "ValueRenderOption", ["formatted", "unformatted", "formula"]
-)("FORMATTED_VALUE", "UNFORMATTED_VALUE", "FORMULA")
-ValueInputOption = namedtuple("ValueInputOption", ["raw", "user_entered"])(
-    "RAW", "USER_ENTERED"
-)
-DateTimeOption = namedtuple(
-    "DateTimeOption", ["serial_number", "formatted_string", "formated_string"]
-)("SERIAL_NUMBER", "FORMATTED_STRING", "FORMATTED_STRING")
-MimeTypeType = namedtuple(
-    "MimeType",
-    ["google_sheets", "pdf", "excel", "csv", "open_office_sheet", "tsv", "zip"],
-)
-MimeType = MimeTypeType(
-    "application/vnd.google-apps.spreadsheet",
-    "application/pdf",
-    "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
-    "text/csv",
-    "application/vnd.oasis.opendocument.spreadsheet",
-    "text/tab-separated-values",
-    "application/zip",
-)
-ExportFormatType = namedtuple(
-    "ExportFormat", ["PDF", "EXCEL", "CSV", "OPEN_OFFICE_SHEET", "TSV", "ZIPPED_HTML"]
-)
-ExportFormat = ExportFormatType(
-    MimeType.pdf,
-    MimeType.excel,
-    MimeType.csv,
-    MimeType.open_office_sheet,
-    MimeType.tsv,
-    MimeType.zip,
-)
 
-PasteType = namedtuple(
-    "PasteType",
-    [
-        "normal",
-        "values",
-        "format",
-        "no_borders",
-        "formula",
-        "data_validation",
-        "conditional_formating",
-    ],
-)(
-    "PASTE_NORMAL",
-    "PASTE_VALUES",
-    "PASTE_FORMAT",
-    "PASTE_NO_BORDERS",
-    "PASTE_FORMULA",
-    "PASTE_DATA_VALIDATION",
-    "PASTE_CONDITIONAL_FORMATTING",
-)
+class Dimension(Enum):
+    rows = "ROWS"
+    cols = "COLUMNS"
 
-PasteOrientation = namedtuple("PasteOrientation", ["normal", "transpose"])(
-    "NORMAL", "TRANSPOSE"
-)
 
-DEPRECATION_WARNING_TEMPLATE = (
-    "[Deprecated][in version {v_deprecated}]: {msg_deprecated}"
-)
+class ValueRenderOption(Enum):
+    formatted = "FORMATTED_VALUE"
+    unformatted = "UNFORMATTED_VALUE"
+    formula = "FORMULA"
+
+
+class ValueInputOption(Enum):
+    raw = "RAW"
+    user_entered = "USER_ENTERED"
+
+
+class DateTimeOption(Enum):
+    serial_number = "SERIAL_NUMBER"
+    formatted_string = "FORMATTED_STRING"
```

Comment:
should we consider including `formated_string` for backwards compatibility? probably not.